src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -147,5 +151,11 @@ public class TagHelperDescriptor
         /// The <see cref="TagHelperUsageDescriptor"/> that indicates how to use the tag helper at design time.
         /// </summary>
         public TagHelperUsageDescriptor UsageDescriptor { get; }
+
+        /// <summary>
+        /// The HTML element the tag helper may output.
+        /// </summary>
+        /// <remarks>Overrides the HTML IntelliSense provided at design time.</remarks>
+        public string OutputElementHint { get; set; }
```

Comment:
Ya that's what I was thinking with 1 & 2 (enforced by the editor) but wanted to present 3 since having conflicting `OutputElementHint`s could potentially indicate that `TagHelper`s will clash. 

3 isn't typically what we've pushed for with other `TagHelper` features since we've normally taken the route of flexibility; but wanted to bring it up since there wasn't a clear path.